integrations/go/workload_executor_test.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/integration/unified"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// hasLoop returns whether or not testCase tc contains a loop operation
+func hasLoop(tc *unified.TestCase) bool {
+	if tc == nil {
+		return false
+	}
+	for _, op := range tc.Operations {
+		if op.Name == "loop" {
+			return true
+		}
+	}
+	return false
+}
+
+func TestAtlasPlannedMaintenance(t *testing.T) {
+	connstring := os.Args[1]
+	workloadSpec := []byte(os.Args[2])
+
+	setupOpts := mtest.NewSetupOptions().SetURI(connstring)
+	if err := mtest.Setup(setupOpts); err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := mtest.Teardown(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// killAllSessions will return an auth error if it's run
+	fileReqs, runners := unified.ParseTestFile(t, workloadSpec, unified.NewOptions().SetRunKillAllSessions(false))
```

Comment:
nit: Is the `runners` variable name a remnant from a previous version of the unified test runner changes where `ParseTestFile` returned a slice of `Runner` instances? It now returns `[]*TestCase`, so `testCases` may be a more fitting name.